pandas/core/series.py

```diff
@@ -4404,9 +4407,22 @@ def map(
         3  I am a rabbit
         dtype: object
         """
-        if callable(arg):
-            arg = functools.partial(arg, **kwargs)
-        new_values = self._map_values(arg, na_action=na_action)
+        if func is None:
+            if "arg" in kwargs:
```

Comment:
Noting this won't be backwards compatible with the use `Series.map(arg=foo, func=myfunc)` where `myfunc` is an argument of `foo`. This seems unlikely enough that I am okay here.

Reply:
You are totally right. In my first implementation I was raising an exception, and then thought it would be better to allow using `arg` as an argument name of the udf. If we want to be very conservative we could raise now, and stop raising when we remove the deprecation and `arg` is no longer accepted as the argument name of the function.

But as you, I think is better to just let that unlikely case break compatibility.